src/components/AppNavigationItem/AppNavigationItem.vue

```diff
@@ -19,92 +19,71 @@
+		<a :class="[ !isIconUrl && bullet === ''  ? icon : '' ]" href="#"
 			@click="callPreventStop(simpleAction, $event)">
-			<img v-if="item.iconUrl" :alt="item.text" :src="item.iconUrl">
-			{{ item.text }}
+			<img v-if="isIconUrl && bullet === ''" :src="icon">
+			{{title}}
 		</a>
 
 		<!-- Popover, counter and button(s) -->
-		<div v-if="item.utils" class="app-navigation-entry-utils">
+		<div v-if="this.$slots.counter || this.$slots.actions" class="app-navigation-entry__utils">
 			<ul>
-				<!-- counter -->
-				<li v-if="Number.isInteger(item.utils.counter) && item.utils.counter > 0"
-					:class="{highlighted: item.utils.counter_highlighted}"
-					class="app-navigation-entry-utils-counter">
-					<span>{{ item.utils.counter }}</span>
-				</li>
-
-				<!-- first action if only one action -->
-				<li v-if="item.utils.actions && item.utils.actions.length === 1"
-					class="app-navigation-entry-utils-menu-button">
-					<button :class="item.utils.actions[0].icon" :title="item.utils.actions[0].text" @click="item.utils.actions[0].action" />
-				</li>
-
-				<!-- menu button if at least two actions -->
-				<li v-else-if="item.utils.actions && item.utils.actions.length > 1"
-					class="app-navigation-entry-utils-menu-button">
-					<button v-click-outside="hideMenu" @click="toggleMenu" />
-				</li>
+				<slot name="counter" />
+				<slot name="actions" />
 			</ul>
 		</div>
 
-		<!-- menu if at least two actions -->
-		<div v-if="item.utils && item.utils.actions && item.utils.actions.length > 1"
-			:class="{ 'open': openedMenu }" class="app-navigation-entry-menu">
-			<PopoverMenu :menu="item.utils.actions" />
-		</div>
-
-		<!-- undo entry -->
-		<div v-if="item.undo" class="app-navigation-entry-deleted">
-			<div class="app-navigation-entry-deleted-description">
-				{{ item.undo.text }}
-			</div>
-			<button :title="t('settings', 'Undo')" class="app-navigation-entry-deleted-button icon-history" />
-		</div>
-
-		<!-- edit entry -->
-		<div v-if="item.edit" class="app-navigation-entry-edit">
-			<form @submit.prevent.stop="item.edit.action">
-				<input :placeholder="item.edit.text" type="text">
-				<input type="submit" value="" class="icon-confirm">
-				<input type="submit" value="" class="icon-close"
-					@click.stop.prevent="cancelEdit">
-			</form>
-		</div>
-
-		<!-- if the item has children, inject the component with proper data -->
-		<ul v-if="item.children">
-			<app-navigation-item v-for="(child, key) in item.children" :key="key" :item="child" />
+		<!-- Children elements -->
+		<ul v-if="canHaveChildren && this.$slots.children ">
+			<slot name='children' />
 		</ul>
 	</nav-element>
 </template>
 
 <script>
-import { PopoverMenu } from 'Components/PopoverMenu'
 import ClickOutside from 'vue-click-outside'
 
 export default {
 	name: 'AppNavigationItem',
 	components: {
-		PopoverMenu
 	},
 	directives: {
 		ClickOutside
 	},
 	props: {
+		title: {
+			type: String,
+			required: true
+
+		},
 		item: {
```

Comment:
The `item` will be finally removed completely, hmm?

Reply:
That's the idea, yes!

Reply:
..Or we could keep it just for the navElement that you implemented..